features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,588 @@
+
+  const setDialogTab = (
+    event: MouseEvent<HTMLElement>,
+    newAlignment: string
+  ) => {
+    setTabIndex(newAlignment);
+  };
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String) => {
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+  if (
+    activeSponsors !== null &&
+    props.selectedSponsor !== null &&
+    activeSponsors[props.selectedSponsor] !== null &&
+    latestPrice !== null &&
+    collReq !== null &&
+    emp !== null &&
+    currentTime !== null &&
+    finalFee !== null &&
+    tokenSymbol !== null &&
+    tokenBalance !== null &&
+    tokenAllowance !== null &&
+    collSymbol !== null &&
+    collBalance !== null &&
+    collAllowance !== null
+  ) {
+    const collBalanceNum = collBalance || 0;
+    const tokenBalanceNum = tokenBalance || 0;
+    const finalFeeNum = finalFee || 0;
+    const collateralToDepositNum = Number(collateralToDeposit) || 0;
+    const maxTokensToLiquidateNum = Number(maxTokensToLiquidate) || 0;
+
+    const sponsorPosition = activeSponsors[props.selectedSponsor];
+
+    const pendingWithdrawTimeRemaining =
+      Number(sponsorPosition.withdrawalTimestamp) - Number(currentTime);
+
+    const pendingTransferTimeRemaining =
+      Number(sponsorPosition.TransferTimestamp) - Number(currentTime);
+
+    const prettyTimeRemaining = (timeRemaining: number) => {
+      return timeRemaining > 0
+        ? Math.max(0, Math.floor(timeRemaining / 3600)) +
+            ":" +
+            Math.max(0, Math.floor((timeRemaining % 3600) / 60)) +
+            ":" +
+            Math.max(0, (timeRemaining % 3600) % 60)
+        : "None";
+    };
+    const pendingTransferTimeString = prettyTimeRemaining(
+      pendingTransferTimeRemaining
+    );
+    const pendingWithdrawTimeString = prettyTimeRemaining(
+      pendingWithdrawTimeRemaining
+    );
+
+    const underCollateralizedPrice =
+      Number(sponsorPosition.collateral) === 0 ||
+      Number(sponsorPosition.tokensOutstanding) === 0 ||
+      Number(collReq) === 0
+        ? 0
+        : Number(sponsorPosition.collateral) /
+          (Number(sponsorPosition.tokensOutstanding) *
+            parseFloat(fromWei(collReq)));
+
+    const underCollateralizedPercent =
+      Number(latestPrice) === 0
+        ? 0
+        : ((Number(latestPrice) - Number(underCollateralizedPrice)) /
+            Number(latestPrice)) *
+          100;
+
+    const collBalanceTooLow = () => {
+      if (tabIndex == "deposit") {
+        return collBalanceNum < collateralToDepositNum;
+      }
+      if (tabIndex == "liquidate") {
+        return collBalanceNum < finalFeeNum;
+      }
+    };
+
+    const tokenBalanceTooLow = tokenBalanceNum < maxTokensToLiquidateNum;
+
+    const needCollateralAllowance = () => {
+      if (tabIndex == "deposit") {
+        if (collAllowance === null || collateralToDeposit === null) return true;
```

Comment:
nit: `collAlllowance` cannot be `null` do to the big `if` statement. And you should replace `collateralToDeposit === null` with `collateralToDepositNum === 0`